src/unseal.js

```diff
@@ -0,0 +1,24 @@
+const axios = require('axios');
+
+const { VAULT_KEYS, VAULT_ADDR } = process.env;
+const KEYS_ARR = VAULT_KEYS.split(',');
+
+function sealStatus() {
+  return axios.get(`${VAULT_ADDR}/v1/sys/seal-status`)
+    .then((res) => res.data.sealed);
+}
+
+function unsealVault(key) {
+  return axios.post(`${VAULT_ADDR}/v1/sys/unseal`, { key });
+}
+
+sealStatus()
+  .then((sealed) => {
+    if (sealed) {
+      console.log('Vault is sealed, unsealing...');
+      KEYS_ARR.forEach(unsealVault);
```

Comment:
I don't think this works as intended in the failure case as you don't return the result of all of the two underlying promises. You'll need to use `Promise.all`. Alternatively, if you have updated to Node 12 you'll be able to use async/await to make it imperative code.

Reply:
I have refactored this, but only based on googling etc, does it make sense now?

Reply:
I think this still doesn't do what we want for the reasons discussed. Here is the refactored snippet. You definitely need to test it works though!

```
async function sealStatus() {
  const response = await axios.get(`${vaultAddress}/v1/sys/seal-status`);
  return res.data;
}

function unsealVault(key) {
  return axios.post(`${vaultAddress}/v1/sys/unseal`, { key });
}

async function unsealChecker() {
  try {
    const sealed = await sealStatus();

    if (sealed) {
      console.log('Vault is currently sealed');
      const unsealPromises = KEYS_ARR.map(unsealVault);
      await Promise.all(unsealPromises);
    } else {
      console.log('Vault is currently unsealed');
    }
  } catch(err) {
    console.log(`Error: ${err}.`);
  }
}

unsealChecker();
```